packages/core/src/tracing/trace.ts

```diff
@@ -12,6 +12,12 @@ import { spanTimeInputToSeconds } from '../utils/spanUtils';
 interface StartSpanOptions extends TransactionContext {
   /** A manually specified start time for the created `Span` object. */
   startTime?: SpanTimeInput;
+
+  /**
+   * Metadata associated with the transaction, for internal SDK use.
+   * @deprecated Use attributes or store data on the scope instead.
+   */
+  metadata?: Partial<TransactionMetadata>;
```

Comment:
Can you explain the thought process behind adding a deprecated field? Also, it looks like `TransactionContext` already has this field. If it's just for a slightly different JSDoc, we should say "with the span" here right?

Reply:
this is already added, but the eslint deprecation rule does not seem to pick up inherited deprecated fields 😬 so without this, it does not show up as deprecated. This is generally very sad...

Reply:
Damn that's crazy/scary. It's fine to keep it then!